### ISSUE TITLE
Port forwarding rules are not restored after rollback

### ISSUE THREAD

**Issue by flavio** [jedi4ever/sahara#49]
How to reproduce the issue:
  * create a Vagrantfile using port forwarding
  * vagrant up => port forwarding works
  * vagrant sandbox on
  * vagrant sandbox commit
  * vagrant halt -f
  * vagrant sandbox rollback => port forwarding rule is not restored

I noticed this behaviour when using the libvirt provider.

**Comment by sciurus** [comment]
Aren'y you missing a step at the end? You ran `vagrant halt`, so the vm is stopped when you ran `vagrant sandbox rollback`. You need to run `vagrant up` afterwards, then the VM should start and port forwarding should be restored.

**Comment by flavio** [comment]
Nope, I took the snapshot while the system was running. Rolling back brings the system online immediately, there's no need to do a `up`.

Taking a snapshot of a powered off system, doing a rollback, doing `up` works like a charm.

I changed the title accordingly.

**Comment by sciurus** [comment]
Port forwarding in vagrant-libvirt starts an SSH process on your host computer. Running "vagrant halt -f" is going to kill its connection to the guest. Vagrant-libvirt should restart the SSH process for port forwarding the next time you run "vagrant up".

**Comment by flavio** [comment]
> Vagrant-libvirt should restart the SSH process for port forwarding the next time you run "vagrant up".

And that's exactly what it does.

The problem is different.

I take a snapshot using sahara while the image is running. Then I halt the image (doesn't matter how). The next time I rollback sahara will automatically restore the system in a **running** state (there's not need to do a `vagrant up`). Now all the port forwarding rules are missing.

I peeked a bit into the code, sahara implements rollback using [fog](https://github.com/fog/fog). libvirt-vagrant is not involved, hence the port forwarding rules are not restored by it.

**Comment by sciurus** [comment]
Exactly.
